internal/tracer.go

```diff
@@ -100,7 +100,7 @@ func (t *tracingContextPropagator) InjectFromWorkflow(
 	if spanContext == nil {
 		return nil
 	}
-	return t.tracer.Inject(spanContext, opentracing.HTTPHeaders, tracingWriter{hw})
+	return t.tracer.Inject(spanContext, opentracing.TextMap, tracingWriter{hw})
```

Comment:
Actually, both may be correct. It's not clear from the context of this function which medium the context is being encoded into. If it's HTTP request, then the old code was correct. If it's some other medium, like storing the serialized context in the database, then both formats are fine, as long as the same format is used when reading and decoding the stored context.

Reply:
Yeah, good point. I think Cadence is using Tchannel/Thrift for its RPCs and this also gets persisted in DB. Like you mentioned, the problem here is inconsistent encoding protocols for reading and writing the context.

Reply:
So does it mean that HTTPHeaders will work for both cases? Then why do we need this change?

Reply:
HTTPHeaders could work for Cadence (though we would then need to change line 110 to make sure we decode using `opentracing.HTTPHeaders` so that we properly URL-decode.

From the opentracing docs:

```
// HTTPHeaders represents SpanContexts as HTTP header string pairs.
	//
	// Unlike TextMap, the HTTPHeaders format requires that the keys and values
	// be valid as HTTP headers as-is (i.e., character casing may be unstable
	// and special characters are disallowed in keys, values should be
	// URL-escaped, etc)
```

I don't think it matters in this case, but using the HTTPHeaders encoding seems to be slightly stricter and may lose data like case-sensitivity while encoding.

Reply:
Got it. Thanks!